### PR TITLE
Being able to retry if an error is found 

### PR DESCRIPTION
**Pull request by PedroCarrillo**
issue #230

added test cases with/without retry and with/without persister. Let me know what can I improve

### CODE REVIEW

File: store/src/test/java/com/nytimes/android/external/store/GetRefreshingErrorTest.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import java.util.concurrent.atomic.AtomicInteger;
+
+import rx.Observable;
+import rx.observers.AssertableSubscriber;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GetRefreshingErrorTest {
```
Comment (digitalbuddha) [inline store/src/test/java/com/nytimes/android/external/store/GetRefreshingErrorTest.java:27]: We should also test this with NetworkBeforeStale and RefreshOnStale caching policies

Comment (PedroCarrillo) [reply store/src/test/java/com/nytimes/android/external/store/GetRefreshingErrorTest.java:27]: ok. will work on the tests

**Comment by digitalbuddha** [comment]
Hows it going?

**Comment by PedroCarrillo** [comment]
I'm having a problem with my tests with a persister where the store.clear does not trigger a network call

**Comment by PedroCarrillo** [comment]
Added tests for NetworkBeforeStale and RefreshOnStale. Sorry it took a little bit to get a grasp of how to test the store with a persister :). Let me know if something I should change/update

**Review by digitalbuddha — APPROVED** [review]
:100:

**Comment by digitalbuddha** [comment]
guess what branch you need to open this against next :-D  Last time,

**Pull request merged**